cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetDataHelper.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2022 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.config;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import com.linkedin.kafka.cruisecontrol.model.Broker;
+import com.linkedin.kafka.cruisecontrol.model.ClusterModel;
+
+/**
+ * This is a helper class to contain helper functions to get brokerSet mappings and reverse mappings
+ */
+public class BrokerSetDataHelper {
+    private final ClusterModel _clusterModel;
+    private final Map<String, List<Integer>> _brokerIdsByBrokerSet;
+
+    public BrokerSetDataHelper(ClusterModel clusterModel, BrokerSetDataStore brokerSetDataStore) {
+        _clusterModel = clusterModel;
+        _brokerIdsByBrokerSet = brokerSetDataStore.getBrokersByBrokerSetId();
+    }
+
+    /**
+     * Helper function to get Broker to BrokerSet mapping
+     *
+     * @return Broker to BrokerSet Map
+     */
+    public Map<String, Set<Broker>> getBrokersByBrokerSet() {
+        return _brokerIdsByBrokerSet.entrySet()
+                .stream()
+                .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue()
+                        .stream()
+                        .map(id -> _clusterModel.broker(id))
+                        .collect(Collectors.toSet())));
+    }
+
+    /**
+     * Helper function to get broker to brokerSet mapping
+     *
+     * @return broker set by broker mapping
+     */
+    public Map<Broker, String> getBrokerSetByBroker() {
+        Map<Broker, String> brokerToBrokerSetMap = new HashMap<>();
+
+        _brokerIdsByBrokerSet.forEach(
+                (brokerSet, brokerIds) -> {
+                    brokerIds.forEach(id -> brokerToBrokerSetMap.put(_clusterModel.broker(id), brokerSet));
+                }
+        );
+
+        return brokerToBrokerSetMap;
+    }
```

Comment:
Computations in both (1) `getBrokersByBrokerSet()` and (2) `getBrokerSetByBroker()` are expensive. Looks like currently they can be called very frequently, and every time they recompute the same data (i.e. this mapping is not expected to change between different calls).

Should we compute them once in `BrokerSetAwareGoal#initGoalState` and cache them in `BrokerSetAwareGoal`? Then maybe we can also drop this helper class.

Reply:
moved to init ``initGoalState`` to reduce computations. These will now be called at each instance of the ``BrokerSetAwareGoal`` class when the ``initGoalState`` is called.